pennylane_lightning/core/src/simulators/lightning_tensor/cutn/base/TensornetBase.hpp

```diff
@@ -0,0 +1,71 @@
+// Copyright 2024 Xanadu Quantum Technologies Inc.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @file TensornetBase.hpp
+ * Base class for all cuTensorNet backends.
+ */
+
+#pragma once
+
+#include <vector>
+
+#include "Error.hpp"
+
+namespace Pennylane::LightningTensor::Cutn {
+template <class Precision, class Derived> class TensornetBase {
+  private:
+    std::size_t numQubits_;
+    std::vector<std::size_t> qubitDims_;
+
+  public:
+    TensornetBase() = delete;
+
+    explicit TensornetBase(const std::size_t numQubits)
+        : numQubits_(numQubits) {
+        qubitDims_.resize(numQubits);
+        std::fill(qubitDims_.begin(), qubitDims_.end(), std::size_t{2});
```

Comment:
Would that work? Or is `resize` better in case it is the same size? 
```suggestion
        qubitDims = std::vector<std::size_t>(numQubits, 2);
```

Reply:
Nice catch!